test/unit/disposable_common.dart

```
@@ -352,6 +351,89 @@ void testCommonDisposable(Func<StubDisposable> disposableFactory) {
+
+      controller.addError(new Exception('intentional'));
+      await new Future(() {});
+
+      expect(disposable.isDisposed, isFalse);
+      expect(disposable.disposalTreeSize, equals(previousTreeSize + 1));
+    });
+
+    group('asFuture should return a future', () {
+      test('that completes when the stream closes', () async {
+        // ignore: close_sinks
+        var controller = new StreamController<Null>();
+        // ignore: cancel_subscriptions
+        var subscription = disposable.listenToStream(controller.stream, (_) {});
+        var future = subscription.asFuture('intentional');
+        await controller.close();
+        var value = await future;
+        expect(value, 'intentional');
+      });
+
+      test('that completes when the stream emits an error', () async {
+        // ignore: close_sinks
+        var controller = new StreamController<Null>();
+        // ignore: cancel_subscriptions
+        var subscription = disposable.listenToStream(controller.stream, (_) {});
+        var future = subscription.asFuture('intentional');
+        controller.addError(new Exception('intentional'));
+        var value = await future;
+        expect(value, null);
+      });
+
+      test(
+          'that un-manages the subscription on completion '
+          'with an error even when cancelOnError is false', () async {
+        var previousTreeSize = disposable.disposalTreeSize;
+
+        // ignore: close_sinks
+        var controller = new StreamController<Null>();
+        // ignore: cancel_subscriptions
+        var subscription = disposable.listenToStream(controller.stream, (_) {},
+            cancelOnError: false, onError: (_, [__]) {});
+        var future = subscription.asFuture('intentional');
+        expect(disposable.disposalTreeSize, equals(previousTreeSize + 1));
+
+        controller.addError(new Exception('intentional'));
+        await new Future(() {});
+        await future;
```

Fixed above by just suppressing the error since that's tested separately.